src/Exception/CompositeException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace React\Promise\Exception;
+
+class CompositeException extends \Exception
```

Comment:
How about `MultiReasonException`?

Reply:
No objections to either class name, but the use of this class appears a bit unclear. Does it make sense to add documentation for this class and/or mark this class a `@internal`?

Reply:
Added documentation in 07b7afd. I think it makes sense to keep it public API is it might be useful for custom collection functions.